addons/core/functions/intercom/fnc_setIntercomChannel.sqf

```diff
@@ -0,0 +1,33 @@
+#include "script_component.hpp"
+
+/*
+  Name: TFAR_fnc_setIntercomChannel
+
+  Author: DartRuffian
+    Sets the channel of a player's vehicle intercom.
+
+  Arguments:
+    0: Vehicle <OBJECT>
+    1: Player <OBJECT>
+    2: Channel <NUMBER>
+
+  Return Value:
+    None
+
+  Example:
+    [vehicle player, player, 1] = call TFAR_fnc_setIntercomChannel;
+
+  Public: Yes
+*/
+
+params [
+    ["_vehicle", objNull, [objNull]],
+    ["_player", objNull, [objNull]],
+    ["_channel", 1, [1]]
+];
+
+if (!alive _vehicle || !alive _player || !isPlayer _player) exitWith {};
+
+_vehicle setVariable [format ["TFAR_IntercomSlot_%1", netId _player], _channel, true];
+["TFAR_intercomChannelSet", [_vehicle, _player, _channel]] call CBA_fnc_localEvent;
```

Comment:
Usually we use tfars event system. Which is a wrapper around CBA.
I don't know if we do it like this elsewhere, it's probably fine?

Reply:
I saw fnc_fireEventHandlers but the function description was "Fires all eventhandlers associated with the passed unit", which seemed to imply it was mainly for units, so I left it as just a CBA event directly.

Reply:
I changed it because I realized that all other TFAR events are `TFAR_event_<name>`, so I just changed it to use the fireEventHandlers function for consistency.

All other events are also named `onX`, so I changed the event name to `onIntercomChannelSet` to match.